Realm/RLMSectionedResults.h

```diff
@@ -0,0 +1,956 @@
+     [tv reloadRowsAtIndexPaths:changes.modifications withRowAnimation:UITableViewRowAnimationAutomatic];
+     [tv insertSections:changes.sectionsToInsert withRowAnimation:UITableViewRowAnimationAutomatic];
+     [tv deleteSections:changes.sectionsToRemove withRowAnimation:UITableViewRowAnimationAutomatic];
+     [tv endUpdates];
+
+ All of the arrays in an `RLMSectionedResultsChange` are always sorted in ascending order.
+ */
+@interface RLMSectionedResultsChange : NSObject
+/// The index paths of objects in the previous version of the collection which have
+/// been removed from this one.
+@property (nonatomic, readonly) NSArray<NSIndexPath *> *deletions;
+/// The index paths in the new version of the collection which were newly inserted.
+@property (nonatomic, readonly) NSArray<NSIndexPath *> *insertions;
+/// The index paths in the old version of the collection which were modified.
+@property (nonatomic, readonly) NSArray<NSIndexPath *> *modifications;
+/// The indices of the sections to be inserted.
+@property (nonatomic, readonly) NSIndexSet *sectionsToInsert;
+/// The indices of the sections to be removed.
+@property (nonatomic, readonly) NSIndexSet *sectionsToRemove;
+/// Returns the index paths of the deletion indices in the given section.
+- (NSArray<NSIndexPath *> *)deletionsInSection:(NSUInteger)section;
+/// Returns the index paths of the insertion indices in the given section.
+- (NSArray<NSIndexPath *> *)insertionsInSection:(NSUInteger)section;
+/// Returns the index paths of the modification indices in the given section.
+- (NSArray<NSIndexPath *> *)modificationsInSection:(NSUInteger)section;
+@end
+
+
+/// The `RLMSectionedResult` protocol defines properties and methods common to both `RLMSectionedResults and RLMSection`
+@protocol RLMSectionedResult <NSFastEnumeration, RLMThreadConfined>
+
+#pragma mark - Object Access
+
+/// The count of objects in the collection.
+@property (nonatomic, readonly) NSUInteger count;
+/// Returns the object for a given index in the collection.
+- (id)objectAtIndexedSubscript:(NSUInteger)index;
+/// Returns the object for a given index in the collection.
+- (id)objectAtIndex:(NSUInteger)index;
+
+#pragma mark - Freeze
+
+/**
+ Returns a frozen (immutable) snapshot of this collection.
+
+ The frozen copy is an immutable collection which contains the same data as this
+ collection currently contains, but will not update when writes are made to the
+ containing Realm. Unlike live arrays, frozen collections can be accessed from any
+ thread.
+
+ @warning This method cannot be called during a write transaction, or when the
+          containing Realm is read-only.
+ @warning Holding onto a frozen collection for an extended period while performing
+          write transaction on the Realm may result in the Realm file growing
+          to large sizes. See `RLMRealmConfiguration.maximumNumberOfActiveVersions`
+          for more information.
+ */
+- (instancetype)freeze;
```

Comment:
If you can freeze an individual section, what happens when a change is made to the sectionedResults that would cause a recalculation of the section keys, and the frozen section would be affected?

Reply:
A frozen section will get a copy of the parent sectioned results, so any changes to the original parent won't do anything to frozen sections. A thawed section will produce a new up to date copy of the sectioned results, and if in the meantime that section has been deleted, the thaw method will throw.